### ISSUE TITLE
`aria-sort`  forces `role="columnheader"` or `rowheader` Validation throws error on `<th>` 

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/aria#592]
The way`aria-sort`  it is written, it is invalid HTML to rely on the `<th>` like this:

`<th aria-sort="acsending">`

There is a thread here:

https://github.com/w3c/html/issues/909

@stevefaulkner 
@sideshowbarker 

Said the issue is with wai-aria. Here's the issue:
* using `aria-sort` requires `role="columnheader"` or `rowheader`
* ARIA currently maps `<th>` to `role="cell"` which provides no appreciable semantics to distinguish a header from a data cell, forcing the author to use `role="columnheader"` or `rowheader` this is an extra step for authors.
* This could be made simpler by introducing an new generic `role="tableheader"`. This would be an optional role to the more specific `rowheader` and `columnheader`. 
* The A11Y API could then map the `<th>` to `role="tableheader"`.
* We could then allow authors to use `<th aria-sort="ascending">` without burdening them with a role, because the `<th>` semantics would be sufficient.

The first rule of ARIA is to use generic HTML when possible. It seems with this minor tweak, we could make it easier for authors to `use aria-sort`. There are no implications on screen readers which already understand what the `th` does. They ignore the unhelpful `role="cell"` mapping in the API.

**Comment by hartman** [comment]
Just wondering, but i'm guessing that [scope](https://www.w3schools.com/tags/att_th_scope.asp) is not really affecting this either is it ? Should it ?

**Comment by hartman** [comment]
And on that note. What about thead ? <th> cells inside a <thead> should be reasonably safe to assume to be column headers right ?

**Comment by DavidMacDonald** [comment]
I think scope doesn't do that much these days, bit its good to know that Safari changes the role to column header instead of the unhelpful `role="cell"`

We "could" say:
* `th` with `scope="col"` map to `role="columnheader"`
* `th` with `scope="row"` map to `role="rowheader"`
* `th` inside `thead` map to `role="columnheader"`

I think we should do that, but that still requires the author to add the `scope` or `thead`.

I think in addition a new role=`tableheader` will benthe most elegant because it would allow the implicit mapping of `th` to that role, and then the author could simply do `<th aria-sort="ascending">`

**Comment by hartman** [comment]
Also relevant: https://www.w3.org/TR/WCAG20-TECHS/H63.html

**Comment by DavidMacDonald** [comment]
We wrote that technique about 15 years ago, in the early days of WCAG. Screen readers have largely overcome the need for scope except in limited situations.

**Comment by jnurthen** [comment]
Looks like HTML-AAM was updated to map `th` to `columnheader` or `rowheader` - closing this
